byterun/alloc.c

```diff
@@ -38,12 +38,14 @@ CAMLexport value caml_alloc (mlsize_t wosize, tag_t tag)
 
   CAMLassert (tag < 256);
   CAMLassert (tag != Infix_tag);
-  if (wosize == 0){
-    result = Atom (tag);
-  }else if (wosize <= Max_young_wosize){
-    Alloc_small (result, wosize, tag);
-    if (tag < No_scan_tag){
-      for (i = 0; i < wosize; i++) Field (result, i) = Val_unit;
+  if (wosize <= Max_young_wosize){
```

Comment:
Pushed `wosize == 0` into the `wosize <= Max_young_wosize` branch.  It can thus be avoided when executing the `caml_alloc_shr` branch.